### PR TITLE
Features achievements

### PR DESCRIPTION
**Pull request by Lemonymous**
Adds native support for achievements to the mod loader.

It works much in the same way as my achievements library.
One change that is of note is that achievements now can have `addReward` and `remReward` functions defined, that will automatically be triggered when an achievement is completed/undone (as well as on init for already completed achievements). This is to make it easier to link rewards to achievements, and have rewards be given/taken away when achievement status changes (for example if a trigger all button were to be added or something similar)

Achievements are now tied to the active profile. All achievements are reset on new profiles, and you can change profiles and have separate achievement progress across them.

Achievements can now be associated with a squad. Squad achievements will be visible in the hangar and in the escape menu while a squad with custom achievement has been selected.
While any number of achievements can be associated with one squad, and they are all visible on the achievement screen for mods, only 3 will be visible in the hangar and in the escape menu. Mod coins associated with the achievements are also displayed on the screen when selecting squads in the hangar or from the statistics screen.

Victory medals are now automatically tracked and displayed for all modded squads in the hangar screen, escape menu, and when selecting a squad in the hangar or from the statistics screen.

When adding squads with `modApi.addSquad`, squads can be given a unique id that can be referenced when adding achievements for the squad. If no id is provided, the mod loader will generate one with the format `mod_id..squad_name`, where all spaces are replaced with underscores.
Example:
`modApi:addSquad({"SquadName","Mech1Name", "Mech2Name", "Mech3Name", id = "SquadId"}, etc.. )`

function list: (incomplete, they will all be further documented in the wiki)
- `table` `modApi.achievements:add(achievement)` -- returns the achievement added
- `table` `modApi.achievements:get()` -- returns all achievements
- `table` `modApi.achievements:get(mod_id)` -- returns all achievements added by mod
- `table` `modApi.achievements:get(mod_id, achievement_id)` -- returns one achievement
- `table` `modApi.achievements:getSquadAchievements()` -- returns all squad achievements
- `table` `modApi.achievements:getSquadAchievements(squad_id)` -- returns all achievements associated with squad
- `void` `modApi.achievements:addProgress(mod_id, achievement_id, objective)` -- adds progress from `objective` table to achievement, and triggers it if complete
- `void` `modApi.achievements:trigger(mod_id, achievement_id)` -- triggers achievement
- `void` `modApi.achievements:trigger(mod_id, achievement_id, flag)` -- triggers/untriggers achievement
- `void` `modApi.achievements:trigger(mod_id, achievement_id, objective)` -- same as `addProgress`
- `bool` `modApi.achievements:isComplete(mod_id, achievement_id)` -- returns `true` if the achievement is completed
- `bool` `modApi.achievements:isProgress(mod_id, achievement_id, objective)` -- returns `true` if achievement status is greater than equal to the passed argument `objective`
- `table` `modApi.achievements:getProgress(mod_id, achievement_id)` -- returns a `table` with the current objective status of the achievement
- `bool` `modApi.achievements:canBeAdded(mod_id, achievement_id)` -- returns `true`before mods have been initialized

### CODE REVIEW

**Comment by KnightMiner** [comment]
As a random comment, `[Skill:GetUnlocked](https://github.com/kartoFlane/ITB-ModLoader/wiki/Skill#skillunlocked) should work great alongside this, overriding it to allow unlocking a new shop weapon when `isComplete` is true.

File: scripts/mod_loader/ui/decorations/deco_frame.lua [hunk 1]
```diff
@@ -99,3 +99,38 @@ end
...
+	
+    drawborder(screen, color, self.rect, bordersize)
+end
+
+function DecoBorder:apply(widget)
+    widget:padding(self.bordersize)
+end
+
+function DecoBorder:unapply(widget)
+    widget:padding(self.bordersize)
```
Comment (kartoFlane) [inline scripts/mod_loader/ui/decorations/deco_frame.lua:135]: `unappply` needs to undo the changes done by `apply`:
```suggestion
    widget:padding(-self.bordersize)
```

**Comment by Lemonymous** [comment]
I've found some more issues in this PR that needs to be dealt with before it can be merged, so it'll need to stay open until I have time to properly test and fix them.

File: scripts/mod_loader/mod_loader.lua [hunk 2]
```diff
@@ -71,6 +71,7 @@ function mod_loader:init()
 		i = i + 1
 		modApi.events.onModMetadataDone:dispatch(id)
 	end
+	modApi.events.onModMetadataDone:unsubscribeAll()
```
Comment (kartoFlane) [inline scripts/mod_loader/mod_loader.lua:74]: Seems this line got duplicated, somehow
